### DISCUSSION TITLE
Does fsspec provide a way to set the content-type when writing a file to azure blob storage?

### DISCUSSION THREAD

**Discussion by pedro-cf** [fsspec/filesystem_spec#1606] [Q&A]
Does fsspec provide a way to set the content-type when writing a file to azure blob storage?

**Comment by martindurant** [comment]
This depends on the backend, each has different specifics.

**Reply by pedro-cf** [reply]
Trying to do it with azure blob storage.

When I upload image files their content type is `application/octet-stream`

I'm using https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/objectstorage.html

**Reply by martindurant** [reply]
OK, so please ask adlfs: https://github.com/fsspec/adlfs/issues/new .

**Reply by pedro-cf** [reply]
Greetings it seems adlfs has a way: https://github.com/fsspec/adlfs/pull/392 but there doesn't seem to be a generic way to solve this.

https://github.com/apache/airflow/issues/39722#issuecomment-2134518387
>  As the content-type is not part of the spec, there is no generic way of setting it. Happy to accept a patch for it, but it is probably more beneficial to all downstream users of fsspec to supply such a patch to fsspec.
